cpp/src/arrow/dataset/file_parquet_encryption_test.cc

```diff
@@ -106,19 +117,65 @@ class DatasetEncryptionTestBase : public ::testing::Test {
     // Write dataset.
     auto dataset = std::make_shared<InMemoryDataset>(table_);
     EXPECT_OK_AND_ASSIGN(auto scanner_builder, dataset->NewScan());
+    // ideally, we would have UseThreads(concurrently) here, but that is not working
+    // unless GH-26818 (https://github.com/apache/arrow/issues/26818) is fixed
```

Comment:
When you say "that is not working", it's simply the `AssertTablesEqual` call that would fail, right?

If that's indeed the case, we can sort the tables before comparing them, to make sure the test is not order-dependent.

Reply:
that is correct

Reply:
Ok, fixed by sorting the tables before comparison.